__tests__/app.test.js

```diff
@@ -109,3 +129,47 @@ it('should return an array of articles' , () => {
   }) 
 })
 
+describe('GET /api/articles/:article_id/comments', () => {
+  it('should return all comments for a given article', () => {
+    return request(app)
+    .get('/api/articles/3/comments')
+    .expect(200)
+    .then(({ body }) => {
+      console.log(body, '<<<<body in test')
+      body.comments.forEach((comment) => {
+        expect(comment).toMatchObject({
+          comment_id: expect.any(Number),
+          votes: expect.any(Number), 
+          created_at: expect.any(String) ,
+          author: expect.any(String) ,
+          body: expect.any(String),
+          article_id: expect.any(Number),
+        })
+      })
+ 
+      expect(body.comments).toBeSortedBy('created_at', {descending: true})
+    })
+  });
+
+it('should respond with 400 error with an invalid id', () => {
+    return request(app)
+    .get('/api/articles/banana/comments')
+    .expect(400)
+    .then(({ body }) => {
+      const { msg } = body;
+      expect(msg).toBe('Invalid input')
+    })
+ })
+
+ it('should respond with 404 error with an id that does not exist', () => {
+  return request(app)
+  .get('/api/articles/999/comments')
+  .expect(404)
+  .then(({ body }) => {
+    const { msg } = body;
+    expect(msg).toBe('article not found')
+  })
+})
+
```

Comment:
Good error testing! 

there's another test you can add - if an article exists but has no comments yet we should get a 200 and an empty array. you will have to check that the article exists!